developer_docs/order_docs.md

```diff
@@ -0,0 +1,427 @@
+
+To create an `order_c ndarray`, we simply copy the flattened data to our `ndarray`'s data placeholder that changes depending on the type.  
+
+If the data is composed of scalars only (ex: `np.array([1, 2, 3])`), an `array_dummy` is created, before copying it to our destination `ndarray`.  
+
+Example:  
+
+```python
+if __name__ == "__main__":
+  import numpy as np
+  a = np.array([[1, 2, 3], [4, 5, 6]])
+```  
+
+Would translate to:
+
+```c
+int main()
+{
+    t_ndarray a = {.shape = NULL};
+    a = array_create(2, (int64_t[]){INT64_C(2), INT64_C(3)}, nd_int64, false, order_c);
+    int64_t array_dummy[] = {INT64_C(1), INT64_C(2), INT64_C(3), INT64_C(4), INT64_C(5), INT64_C(6)}; // Creation of an array_dummy containing the scalars, notice the data is flattened
+    memcpy(a.nd_int64, array_dummy, 6 * a.type_size); // Copying from array_dummy to our ndarray 'a'
+    free_array(a);
+    return 0;
+}
+```  
+  
+If the data is composed of at least one variable array (like `c` in the example below), we would use a series of copy operations to our `
+ndarray`.  
+
+Example:  
+
+```python
+if __name__ == "__main__":
+  import numpy as np
+  a = np.array([1, 2, 3])
+  b = np.array([4, 5, 6])
+  c = np.array([a, [7, 8, 9], b])
+```  
+
+Would translate to this:  
+
+```c
+int main()
+{
+    t_ndarray a = {.shape = NULL};
+    t_ndarray b = {.shape = NULL};
+    t_ndarray c = {.shape = NULL};
+    a = array_create(1, (int64_t[]){INT64_C(3)}, nd_int64, false, order_c);
+    int64_t array_dummy[] = {INT64_C(1), INT64_C(2), INT64_C(3)};
+    memcpy(a.nd_int64, array_dummy, 3 * a.type_size);
+    b = array_create(1, (int64_t[]){INT64_C(3)}, nd_int64, false, order_c);
+    int64_t array_dummy_0001[] = {INT64_C(4), INT64_C(5), INT64_C(6)};
+    memcpy(b.nd_int64, array_dummy_0001, 3 * b.type_size);
+    
+    // 'c' ndarray creation starts here, 'c' is [a, [7, 8, 9], b]
+    
+    c = array_create(2, (int64_t[]){INT64_C(3), INT64_C(3)}, nd_int64, false, order_c); // Allocating 'c' ndarray
+    uint32_t offset = 0; // Initializing offset, used later to avoid overwritting data when executing multiple copy operations
+    array_copy_data(&c, a, offset); // Copying the first element of 'c', 'offset' is 0 since it's our first copy operation
+    offset += a.length; // Incrementing offset for upcoming copy operation
+    int64_t array_dummy_0002[] = {INT64_C(7), INT64_C(8), INT64_C(9)}; // Creating an array_dummy with 'c''s second element's scalars ([7, 8, 9])
+    memcpy(c.nd_int64 + offset, array_dummy_0002, 3 * c.type_size); // 'offset' is also with 'memcpy'
+    offset += 3; // incrementing 'offset', preparing for final copy
+    array_copy_data(&c, b, offset); // Copying the third element to 'c' ndarray
+    free_array(a);
+    free_array(b);
+    free_array(c);
+    return 0;
+}
+```  
+
+### `order_f` array creation example
+
+If the data is one dimensional, all we would need is one copy operation, same as an `order_c ndarray`.
```

Comment:
1D objects always have `order=None` in pyccel (in numpy `c_order = True` and `f_order = True`)

Reply:
Okey, I'll mention this and remove those lines.
For C code, we pass `order_c` to `array_create` in this situation, is that okey?
We could change it for organizational purposes, am not sure if that is very useful

Reply:
https://github.com/pyccel/pyccel/pull/1276/commits/1d6a5d7eb2baa0efcca3d1818bb257cd4656323b
Should I mention that `array_create` default to `order_c` as a parameter if the array is one dimensional?

Reply:
> For C code, we pass order_c to array_create in this situation, is that okey?

I think so. If we never check it then it doesn't matter. `order_c` just means "last dimension is contiguous" so that is true for a 1D array

> We could change it for organizational purposes, am not sure if that is very useful

I agree I don't think it matters

> Should I mention that array_create default to order_c as a parameter if the array is one dimensional?

I wouldn't. As it is a default parameter the developer doesn't need to know this.